packages/test/test-version-utils/src/compatUtils.ts

```diff
@@ -56,98 +50,48 @@ import { getRequestedVersion } from "./versionUtils.js";
-				},
-		enableGroupedBatching = true,
-		enableRuntimeIdCompressor = "on",
-		// Some t9s tests timeout with small settings. This is likely due to too many ops going through.
-		// Reduce chunking cut-off for such tests.
-		chunkSizeInBytes = driverType === "local" ? 200 : 1000,
-	} = options;
-
-	if (version.startsWith("1.")) {
-		options = {
-			// None of these features are supported by 1.3
-			compressionOptions: disabledCompressionConfig,
-			enableGroupedBatching: false,
-			enableRuntimeIdCompressor: undefined,
-			// Enable chunking.
-			// We need to ensure that 1.x documents (that use chunking) can still be opened by 2.x.
-			// This options does nothing for 2.x builds as chunking is only enabled if compression is enabled.
-			chunkSizeInBytes,
-			...options,
-		};
-	} else if (version.startsWith("2.0.0-rc.1.")) {
-		options = {
-			compressionOptions: compressorDisabled, // Can't use compression, need https://github.com/microsoft/FluidFramework/pull/20111 fix
-			enableGroupedBatching,
-			enableRuntimeIdCompressor: undefined, // it was boolean in RC1, switched to enum in RC2
-			chunkSizeInBytes: Number.POSITIVE_INFINITY, // disabled, need https://github.com/microsoft/FluidFramework/pull/20115 fix
-			...options,
-		};
-	} else if (version.startsWith("2.0.0-rc.2.")) {
-		options = {
-			compressionOptions: compressorDisabled, // Can't use compression, need https://github.com/microsoft/FluidFramework/pull/20111 fix
-			enableGroupedBatching,
-			// control over schema was generalized in RC3 - see https://github.com/microsoft/FluidFramework/pull/20174
-			// IdCompressor settings moved around - can't enable them across versions without tripping on asserts
-			enableRuntimeIdCompressor: undefined,
-			chunkSizeInBytes: Number.POSITIVE_INFINITY, // disabled, need https://github.com/microsoft/FluidFramework/pull/20115 fix
-			...options,
-		};
-	} else {
-		// "2.0.0-rc.3." ++
-		options = {
-			compressionOptions,
-			enableGroupedBatching,
-			chunkSizeInBytes,
-			enableRuntimeIdCompressor,
-			...options,
-		};
+function getMinVersionForCollab(
+	runtimeVersion: string,
+	runtimeVersionForLoading: string | undefined,
+): MinimumVersionForCollab {
+	isMinimumVersionForCollab(runtimeVersion);
+	if (runtimeVersionForLoading === undefined) {
+		// If `containerRuntimeForLoading` is not defined, then this is not a cross-client compat scenario.
+		// In this case, we can use the `runtimeVersion` as the default minVersionForCollab.
+		return runtimeVersion;
 	}
+	isMinimumVersionForCollab(runtimeVersionForLoading);
+	// If `containerRuntimeForLoading` is defined, we will use the lower of the two versions to ensure
+	// compatibility between the two runtimes.
+	return semver.compare(runtimeVersion, runtimeVersionForLoading) <= 0
+		? runtimeVersion
+		: runtimeVersionForLoading;
+}
 
-	return options;
+/**
+ * Asserts the given version is valid semver and is type MinimumVersionForCollab.
+ */
+function isMinimumVersionForCollab(
```

Comment:
I found this function name a bit confusing. Before getting to its definition and its TSDoc, I thought it was supposed to return whether a particular version is a minimum of some kind, which didn't make sense since it was the only parameter and thus apparently wasn't being compared to something else.

Maybe the type should be just `VersionForCollab`, and the function `isVersionForCollab`?

Reply:
Maybe `valideTypeIsMinVersionForCollab`